app/src/main/java/com/google/samples/apps/sunflower/compose/plantdetail/PlantDetailView.kt

```diff
@@ -187,26 +182,91 @@ private fun PlantDetailsContent(
     plant: Plant,
     isPlanted: Boolean,
     onNamePosition: (Float) -> Unit,
-    callbacks: PlantDetailsCallbacks,
+    onFabClick: () -> Unit,
     contentAlpha: State<Float>
 ) {
     Column(Modifier.verticalScroll(scrollState)) {
-        PlantImageHeader(
-            scrollState, plant.imageUrl, callbacks.onFabClick, isPlanted,
-            Modifier.graphicsLayer(contentAlpha.value)
-        )
-        PlantInformation(
-            name = plant.name,
-            wateringInterval = plant.wateringInterval,
-            description = plant.description,
-            onNamePosition = { onNamePosition(it) },
-            toolbarState = toolbarState
+        ConstraintLayout {
+            val (image, fab, info) = createRefs()
+
+            PlantImage(
+                imageUrl = plant.imageUrl,
+                modifier = Modifier
+                    .constrainAs(image) { top.linkTo(parent.top) }
+                    .alpha(contentAlpha.value)
+            )
+
+            if (!isPlanted) {
+                val fabEndMargin = Dimens.PaddingSmall
+                PlantFab(onFabClick = onFabClick, modifier = Modifier
+                    .constrainAs(fab) {
+                        centerAround(image.bottom)
+                        absoluteRight.linkTo(parent.absoluteRight, margin = fabEndMargin)
+                    }
+                    .alpha(contentAlpha.value)
+                )
+            }
+
+            PlantInformation(
+                name = plant.name,
+                wateringInterval = plant.wateringInterval,
+                description = plant.description,
+                onNamePosition = { onNamePosition(it) },
+                toolbarState = toolbarState,
+                modifier = Modifier.constrainAs(info) {
+                    top.linkTo(image.bottom)
+                }
+            )
+        }
+    }
+}
+
+
+@Composable
+private fun PlantImage(
+    imageUrl: String,
+    modifier: Modifier = Modifier,
+    placeholderColor: Color = MaterialTheme.colors.onSurface.copy(0.2f)
+) {
+    CoilImage(
+        data = imageUrl,
+        contentScale = ContentScale.Crop,
+        fadeIn = true,
+        contentDescription = null,
+        loading = {
+            Box(modifier = Modifier
+                .fillMaxSize()
+                .background(placeholderColor))
+        },
+        modifier = modifier
+            .fillMaxWidth()
+            .preferredHeight(Dimens.PlantDetailAppBarHeight)
+    )
+}
+
+@Composable
+private fun PlantFab(
+    onFabClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val fabAccessibilityLabel = stringResource(R.string.add_plant)
```

Comment:
Ditto the chat, move this to the icon below

Reply:
Done